mm2src/adex_cli/src/tests/mod.rs

```diff
@@ -0,0 +1,257 @@
+    let args = vec!["adex-cli", "balance", "RICK"];
+    Cli::execute(args.iter().map(|arg| arg.to_string()), &config, &response_handler)
+        .await
+        .unwrap();
+
+    let result = String::from_utf8(buffer).unwrap();
+    assert_eq!(RICK_BALANCE, result);
+}
+
+#[tokio::test]
+async fn test_enable() {
+    tokio::spawn(fake_mm2_server(7790, "src/tests/enable.http"));
+    test_activation_scheme().await;
+    tokio::time::sleep(Duration::from_millis(FAKE_SERVER_WARMUP_TIMEOUT_MS)).await;
+    let mut buffer: Vec<u8> = vec![];
+    let response_handler = ResponseHandlerImpl {
+        writer: (&mut buffer as &mut dyn Write).into(),
+    };
+    let config = AdexConfigImpl::new("dummy", "http://127.0.0.1:7790");
+    let args = vec!["adex-cli", "enable", "ETH"];
+    Cli::execute(args.iter().map(|arg| arg.to_string()), &config, &response_handler)
+        .await
+        .unwrap();
+
+    let result = String::from_utf8(buffer).unwrap();
+    assert_eq!(ENABLE_OUTPUT, result);
+}
+
+async fn test_activation_scheme() {
+    init_activation_scheme().await.unwrap();
+    let scheme = get_activation_scheme().unwrap();
+    let kmd_scheme = scheme.get_activation_method("KMD");
+    assert!(kmd_scheme.is_some());
+    let kmd_scheme = kmd_scheme.unwrap();
+    assert_eq!(kmd_scheme.get("method").unwrap().as_str().unwrap(), "electrum");
+    assert_eq!(kmd_scheme.get("servers").unwrap().as_array().unwrap().iter().count(), 3);
+}
+
+#[tokio::test]
+async fn test_buy_morty_for_rick() {
+    tokio::spawn(fake_mm2_server(7791, "src/tests/buy.http"));
+    tokio::time::sleep(Duration::from_millis(FAKE_SERVER_WARMUP_TIMEOUT_MS)).await;
+    let mut buffer: Vec<u8> = vec![];
+    let response_handler = ResponseHandlerImpl {
+        writer: (&mut buffer as &mut dyn Write).into(),
+    };
+    let config = AdexConfigImpl::new("dummy", "http://127.0.0.1:7791");
+    let args = vec!["adex-cli", "buy", "MORTY", "RICK", "0.01", "0.5"];
+    Cli::execute(args.iter().map(|arg| arg.to_string()), &config, &response_handler)
+        .await
+        .unwrap();
+
+    let result = String::from_utf8(buffer).unwrap();
+    assert_eq!("Buy order uuid: 4685e133-dfb3-4b31-8d4c-0ffa79933c8e\n", result);
+}
+
+async fn fake_mm2_server(port: u16, response_path: &'static str) {
+    let server = TcpListener::bind(("0.0.0.0", port))
+        .await
+        .expect("Failed to bind tcp server");
+
+    if let Ok((stream, _)) = server.accept().await {
+        tokio::spawn(handle_connection(stream, response_path));
+    }
+}
+
+async fn handle_connection(mut stream: TcpStream, response_path: &'static str) {
+    let mut file = File::open(response_path).unwrap();
+    let mut buffer: Vec<u8> = vec![];
+    file.read_to_end(&mut buffer).unwrap();
+    let (reader, mut writer) = stream.split();
+    reader.readable().await.unwrap();
+    writer.write_all(&buffer).await.unwrap();
+    tokio::time::sleep(Duration::from_millis(FAKE_SERVER_COOLDOWN_TIMEOUT_MS)).await;
```

Comment:
If the path is statically provided, you can ensure it's validity on the compile time with using https://doc.rust-lang.org/std/macro.include_bytes.html instead of loading it on runtime

Reply:
[done](https://github.com/KomodoPlatform/atomicDEX-API/pull/1768/commits/95b2b3e84ad9144681831d0895561dc4cb04e712)